### PR TITLE
ENH: RIX Optics Classes

### PR DESCRIPTION
**Pull request by jsheppard95**
## Description
- Renamed `mirror.XOffsetMirror2` to `mirror.XOffsetMirrorBend` for clarity
- Added `mirror.XOffsetMirrorSwitch` for control of device MR1K2-SWITCH
   - This is nearly identical to `mirror.XOffsetMirror`, but with vertical axis PVs `MMS:YLEFT`/`MMS:YRIGHT` instead of `MMS:YUP`/`MMS:YDWN`
   - These changes are not ideal. It would be best to change the PVs of all mirror gantry axes to be Y1/Y2 for master/slave, allowing for the same parent class of all systems. This requires changes to each system in `lcls-plc-lfe-optics`, `lcls-plc-tmo-optics`, `lcls-plc-txi-optics`, and `lcls-plc-rixs-optics`, lots of work
   - As a first step, it would be ideal to refactor parent class `mirror.XOffsetMirror` so that child classes do not need to set unused components to `None`.

## Motivation and Context
- Python control of RIX optics MR1K1-BEND, MR1K2-SWITCH, and SP1K1-MONO

## How Has This Been Tested?
- Local testing suite
- Created `typhos` screens locally using `$ typhos mirror.XOffsetMirrorSwitch[{"prefix": "MR1K2:SWITCH"}]`, moved vertical axes without issue
- Also created local screen for Mono, but not motion testing yet

## Where Has This Been Documented?
- Docstrings
- Release notes

## Pre-merge checklist
- [X] Code works interactively
- [X] Code contains descriptive docstrings, including context and API
- [ ] New/changed functions and methods are covered in the test suite where possible
- [X] Test suite passes locally
- [x] Test suite passes on travis
- [X] Ran docs/pre-release-notes.sh and created a pre-release documentation page
- [X] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

**Review by sfsyunusSLAC — APPROVED** [review]

**Comment by sfsyunusSLAC** [comment]
Looks good to me! I have an issue open regarding making a base class which just fell off my list and I forgot to do it... I'll work on this next week Issue #659

File: pcdsdevices/spectrometer.py
Comment (klauer) [inline pcdsdevices/spectrometer.py:232]: Would you mind changing these from comments to `doc=''` information?
```suggestion
    m_pi = Cpt(BeckhoffAxis, ':MMS:M_PI', kind='normal', doc='mirror pitch [urad]')
```

It'll show up in the documentation this way.

Also, these subdevices shouldn't be "hinted" (a brief comparison on the kinds is here https://confluence.slac.stanford.edu/display/PCDS/How+to+add+a+device+to+pcdsdevices#Howtoaddadevicetopcdsdevices-Choosingtheright%22kind%22)

I think I'd suggest changing almost all of these to "normal".

**Review by klauer — CHANGES_REQUESTED** [review]
Overall this is good and clean - but I'd appreciate if you could fix up the couple things I mentioned.

File: pcdsdevices/mirror.py
Comment (jsheppard95) [reply pcdsdevices/mirror.py:327]: Yes it is used within happi:
```
(dev) [sheppard@psbuild-rhel7-01  ~]$ happi search name=mr1k1_bend
+--------------------+-------------------------------------------+
| EntryInfo          | Value                                     |
+--------------------+-------------------------------------------+
| active             | True                                      |
| args               | ['{{prefix}}']                            |
| beamline           | KFE                                       |
| creation           | Thu Oct 29 15:21:45 2020                  |
| detailed_screen    | None                                      |
| device_class       | pcdsdevices.mirror.XOffsetMirror2         |
| documentation      | None                                      |
| embedded_screen    | None                                      |
| engineering_screen | None                                      |
| functional_group   | Mirrors                                   |
| kwargs             | {'name': '{{name}}'}                      |
| last_edit          | Thu Oct 29 15:44:04 2020                  |
| lightpath          | False                                     |
| location_group     | Section 07                                |
| macros             | None                                      |
| name               | mr1k1_bend                                |
| parent             | None                                      |
| prefix             | MR1K1:BEND                                |
| prefix_xy          | None                                      |
| stand              | K1S06                                     |
| system             | beam control                              |
| type               | pcdsdevices.happi.containers.OffsetMirror |
| xgantry_prefix     | None                                      |
| z                  | 733.77200                                 |
+--------------------+-------------------------------------------+
```
Though I believe this is the only place. @sfsyunusSLAC do you know of anywhere else?

Comment (klauer) [reply pcdsdevices/mirror.py:327]: There's a slight issue here - we have two primary environments:
1. Development (`dev_conda`)
2. Latest released environment (`pcds_conda`)

If we merge this PR, update the the dev environment, and update happi:

`happi` -> `XOffsetMirrorBend`
`dev` -> `XOffsetMirrorBend`
`pcds_conda` -> `XOffsetMirror2` (... until pcdsdevices is tagged and released, which might be a while)

That's why renaming can be an issue - however, if you make a backward-compatible line like this, it should be fine for both instances:

```python
XOffsetMirror2 = XOffsetMirrorBend
```
(after the class definition, of course)

Comment (jsheppard95) [reply pcdsdevices/mirror.py:343]: Oh sure that makes sense, I can add this line in my next commit. I'm having other issues in the previous one that I'll describe below

File: pcdsdevices/spectrometer.py
Comment (jsheppard95) [reply pcdsdevices/spectrometer.py]: Thanks for sharing the docs on the different component types, that was never clear to me. I added these doc input variables to the components in `spectrometer.Mono` and `mirror.XOffsetMirror`, but now I'm seeing something strange:

When I run the local test suite I'm now getting 4 errors (`test_evr.py`, `test_interface.py`, `test_lens.py`, and `test_mpod.py`), all of which fail with `ImportError: cannot import name 'be_lens_calcs'`.

I then also found I now have the hidden file in my checkout, `pcdsdevices/.nfs24aa81ac87ba001d0000034d`, and if I try to remove it, I get:
`rm: cannot remove ‘pcdsdevices/.nfs24aa81ac87ba001d0000034d’: Device or resource busy`.

Have you seen this sort of thing before? I'm surprised that editing mirror.py and spectrometer.py had an effect on be_lens_calcs

Comment (klauer) [reply pcdsdevices/spectrometer.py]: The `be_lens_calcs` comes from a library `pcdscalc`. If you test using pcds-3.4.1 and up, it should be included. I think it may not be present in the dev environment at the moment - we'll have to add it.

That `.nfs*` thing looks like a temporary file from the file system (NFS). I have seen it before, but not sure in what cases they are generated. You shouldn't worry about it too much, though.

File: pcdsdevices/mirror.py
Comment (ZLLentz) [reply pcdsdevices/mirror.py:343]: Ditto on this- it's the only comment from my review. Without this backcompatibility alias this is actually an API break.

Comment (sfsyunusSLAC) [reply pcdsdevices/mirror.py:343]: m1k1_bend should be the only instance

Comment (jsheppard95) [reply pcdsdevices/mirror.py:343]: Resolved in commit 6d33e1a

File: pcdsdevices/spectrometer.py
Comment (jsheppard95) [reply pcdsdevices/spectrometer.py]: Hmm, I've tried with both pcds-3.4.1 and the dev environment but still getting that issue. Though interestingly I can import `pcdscalc` in a python session with the latest environment.

And yes the `.nfs*` files is now gone.

Comment (ZLLentz) [reply pcdsdevices/spectrometer.py]: Usually `.nfs*` files are caches for hanging processes. They go away when the associated process dies.

I managed to forget to install `pcdscalc` into `dev_conda`, I'm going to do that now.

File: pcdsdevices/mirror.py
Comment (ZLLentz) [reply pcdsdevices/mirror.py:343]: backcompat change looks good to me

**Review by ZLLentz — APPROVED** [review]
Just waiting on the CI to work again

**Comment by ZLLentz** [comment]
I'm going to close and re-open this PR to get it into the travis queue (not that this queue is super slow with Travis's new extortion scheme)

File: pcdsdevices/spectrometer.py
Comment (jsheppard95) [reply pcdsdevices/spectrometer.py]: Sorry I missed the follow up comments on this. Yes I just checked again and the `.nfs*` file is gone and the tests now pass locally again using `dev_conda`, thanks Zach!

**Comment by jsheppard95** [comment]
Any other changes needed before merging this PR? It looks like the goal is to do the Epics checkouts for the RIX optics during this week's PAMM.

**Review by klauer — APPROVED** [review]
Looks great. I'll merge this so you have it in time for the PAMM.

**Pull request merged**
